### DISCUSSION TITLE
Are Bi-Directional Relationships on the Roadmap?

### DISCUSSION THREAD

**Discussion by benjamin-p-newton** [payloadcms/payload#5073] [General]
Hi guys,

Just wondering when/if bi-directional relationships are on the roadmap for the near future?
I consider it a fundamental feature that should be prioritized and the lack of this feature is preventing me from migrating a number of projects to Payload.

Regards,

**Comment by jmikrut** [comment]
Hey man,

Yes! We talked about them for about an hour today in the office. We will get this more formally announced soon, but this is definitely a high priority for us. Do you mind sharing any information about your use case? It would be good to know so that we can plan accordingly!

**Comment by w41x** [comment]
I created a prototype for a plugin. Check it out:

https://github.com/w41x/payload-bidirectional-relationships-plugin

**Comment by tyteen4a03** [comment]
Just as a quick update: In the latest [Payload community call](https://youtu.be/VWMLYb-3n1s), the team has confirmed that bi-directional relationships (including relationships contexts) is coming soon to 3.0!

**Comment by emwadde** [comment]
Hi, we are exploring backend tools for a reasonably large scale web application and built-in two-way relationships is a must have feature for us. Any idea when we might be able to get this feature. We have about a month before we get our hands dirty.
Payloadcms really looks awesome for us

**Reply by JeremieLeblanc** [reply]
Bi-directional relationships have been added to Palyload v3 in the form of the join field: https://payloadcms.com/docs/beta/fields/join

**Comment by GermanJablo** [comment]
Marking as complete. As mentioned, this feature was solved with our join field. Thanks for your feedback!
